Amazon.QLDB.Driver/transaction/AsyncTransaction.cs

```diff
@@ -15,43 +15,164 @@ namespace Amazon.QLDB.Driver
 {
     using System;
     using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
     using Amazon.IonDotnet.Tree;
+    using Amazon.QLDBSession.Model;
+    using Amazon.Runtime;
     using Microsoft.Extensions.Logging;
 
-    internal class AsyncTransaction
+    /// <summary>
+    /// Implementation of a QLDB transaction which also tracks child Results for the purposes of managing their lifecycle.
+    /// Any unexpected errors that occur when calling methods in this class should not be retried, as the state of the
+    /// transaction is now ambiguous. When an OCC conflict occurs, the transaction is closed.
+    ///
+    /// Child Result objects will be closed when the transaction is aborted or committed.
+    /// </summary>
+    internal class AsyncTransaction : BaseTransaction
     {
-        internal AsyncTransaction(Session session, string txnId, ILogger logger, CancellationToken cancellationToken)
+        private CancellationToken cancellationToken = default;
```

Comment:
Nit: the default isn't necessary.

Reply:
Removed default and also made the field readonly.